agent/consul/leader_connect_test.go

```diff
@@ -175,6 +175,12 @@ func waitForActiveCARoot(t *testing.T, srv *Server, expect *structs.CARoot) {
 	})
 }
 
+func getCAProvider(s *Server) (ca.Provider, *structs.CARoot) {
```

Comment:
`getCAProvider` must only be called while holding the `caProviderReconfigurationLock` which is what this helper is doing.

Reply:
Maybe it would be clearer if the test helper was named differently than the method call it wraps, and in a way that explains why it needs to exist?

Reply:
renamed to `getCAProviderWithLock`. That better?